### PR TITLE
Fix crash in EpilogueArcAnalysis

### PR DESCRIPTION
**Pull request by meg-gupta**
EpilogueARCState for unreachable blocks is non-existent. Fix the `EpilogueARCContext::getState` method and it's users for such blocks.

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test windows platform

File: include/swift/SILOptimizer/Analysis/EpilogueARCAnalysis.h [hunk 1]
```diff
@@ -83,8 +83,11 @@ class EpilogueARCContext {
   /// The exit blocks of the function.
   llvm::SmallPtrSet<SILBasicBlock *, 2> ExitBlocks;
 
-  EpilogueARCBlockState &getState(SILBasicBlock *BB) {
-    return IndexToStateMap[*PO->getPONumber(BB)];
+  Optional<EpilogueARCBlockState *> getState(SILBasicBlock *BB) {
+    auto poNumber = PO->getPONumber(BB);
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply include/swift/SILOptimizer/Analysis/EpilogueARCAnalysis.h:88]: Done

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: include/swift/SILOptimizer/Analysis/EpilogueARCAnalysis.h [hunk 2]
```diff
@@ -83,8 +83,12 @@ class EpilogueARCContext {
   /// The exit blocks of the function.
   llvm::SmallPtrSet<SILBasicBlock *, 2> ExitBlocks;
 
-  EpilogueARCBlockState &getState(SILBasicBlock *BB) {
-    return IndexToStateMap[*PO->getPONumber(BB)];
+  Optional<EpilogueARCBlockState *> getState(SILBasicBlock *BB) {
```
Comment (gottesmm) [inline include/swift/SILOptimizer/Analysis/EpilogueARCAnalysis.h:86]: I meant, can you add this in a comment on getState(). A doxygen comment that explains why getState may fail.

**Review by gottesmm — APPROVED** [review]
One last comment fixup. Then I am happy. I am cool if you want to do it in a follow on commit.

File: include/swift/SILOptimizer/Analysis/EpilogueARCAnalysis.h (hunk 2 above)
Comment (meg-gupta) [reply include/swift/SILOptimizer/Analysis/EpilogueARCAnalysis.h:88]: Done : )

**Comment by meg-gupta** [comment]
@swift-ci smoke test and merge

**Comment by meg-gupta** [comment]
@swift-ci smoke test and merge

**Pull request merged**